src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -3627,6 +3638,11 @@ NetworkOPsImp::addAccountHistoryJob(SubAccountHistoryInfoWeak subInfo)
                             *stTxn, meta->getResultTER(), true, curTxLedger);
                         jvTx[jss::meta] = meta->getJson(JsonOptions::none);
                         jvTx[jss::account_history_tx_index] = txHistoryIndex--;
+
+                        if (i + 1 == num_txns ||
```

Comment:
I am not sure if `i + 1 == num_txns` always is a boundary. Please see `struct AccountTxMarker`. 
https://github.com/XRPLF/rippled/blob/e5275b857752c2d6078cb8774edbb7e60f179d72/src/ripple/app/rdb/RelationalDatabase.h#L58